### DISCUSSION TITLE
Customisation of community plugins

### DISCUSSION THREAD

**Discussion by kaka-ruto** [AstroNvim/astrocommunity#327] [Ideas]
I was setting up neotest using the [community setup](https://github.com/AstroNvim/astrocommunity/blob/2a46002e9cace0646b68689965ad4a5a47d23621/lua/astrocommunity/test/neotest/init.lua).

The author has specified use of plugins that I do not intend to use. 

I have customized the installation to include my own plugins, as seen below, but it still installs the listed plugins on astrocommunity.

```lua
  { import = "astrocommunity.test.neotest" },
  {
    "nvim-neotest/neotest",
    dependencies = { "olimorris/neotest-rspec", "zidhuss/neotest-minitest" },
    ft = { "ruby" },
    opts = function(_, opts)
			return utils.extend_tbl(opts, {
				adapters = { require("neotest-rspec"), require("neotest-minitest"), },
			})
		end,
  },
}
```

And the screenshot below showing my and the original author's plugins
<img width="237" alt="Screenshot 2023-06-28 at 17 06 14" src="https://github.com/AstroNvim/astrocommunity/assets/16320124/d1eb06b3-56b0-4e28-b7fc-c22124fe60b3">

Is there a way we can let authors not specify otherwise personal plugins or keymaps, but specify that x and y have to be set up in z manner. Then when installing the community version, and the user hasn't set up the said options, we show a friendly validation error and perhaps a link to the docs showing how to do it?

The other alternative would be to have a flag that lets the user completely overwrite upstream configuration, and not just add to it?

**Comment by Uzaaft** [comment]
You can explicitly disable the plugins you don't need. Read the lazy-docs for the syntax for enabling/disabling plugins: https://github.com/folke/lazy.nvim
